shaders/clouds.omwfx

```diff
@@ -513,9 +502,25 @@ fragment main {
             * max(sun_dir.z, 0.0) * 3.0
             // Regulate direct lighting with max_dist to prevent it appearing on top of close objects
             * min(max_dist * (1.0 / 100000.0), 1.0);
-        vec3 fog_color = (sun_light + moon_light) * mix(sky_color, sky_tone, max(sun_dir.z, 0.3));
-
-        vec3 fog_light = fog_color + direct_light * sun_color;
+        //Define the atmospheric attenuation factors for sunlight's red, green, and blue channels
+        vec3 att_factor = vec3(
+            (pow((2 / (abs(sun_dir.z) + 1)), 2) / 2),
+            (sqrt(1 / (abs(sun_dir.z))) / 10 * (abs(sun_dir.z))),
+            (pow((1 / (abs(sun_dir.z) + 1)), 2) * 3 * (-1 * (abs(sun_dir.z))))
+        );
+        //Interpolate between original sun color and attenuated color based on sun's proximity to the horizon
+        vec3 sun_attenuated = vec3(mix(vec3(min((sun_color.x * ((dot(dir, sun_dir) + 1) / 2) * att_factor.x), 255), min((sun_color.y * (dot(dir, sun_dir) + 1) * att_factor.y), 255), min((sun_color.z / (dot(dir, sun_dir) + 2) * att_factor.z),255)), sun_color, abs(sun_dir.z)));
```

Comment:
What's that 255 doing there? Is that capping the light level? If so, assuming that the maximum value of a colour channel is 255 is a bad assumption: better to simply cap to 1.0 after calculations are done.

Reply:
cap the channels or the whole vector?